### PR TITLE
Rename `BA2026.EnableAdditionalSdlSecurityChecks`

### LINKED ISSUE #584: Rename `BA2026.EnableAdditionalSdlSecurityChecks` to `BA2026.EnableMicrosoftCompilerSdlSwitch`
The existing rule name sounds extremely generic. The new rule name is intended to make it a bit clearer that this rule is enforcing a specific Microsoft compiler security option, its ['/sdl'](https://docs.microsoft.com/en-us/cpp/build/reference/sdl-enable-additional-security-checks?view=msvc-170) switch.

### PR DESCRIPTION
**Pull request by marmegh**
Fix #584.

`EnableAdditionalSdlSecurityChecks` is unnecessarily vague. Renaming to `EnableMicrosoftCompilerSdlSwitch` to increase clarity for customers.

### CODE REVIEW

File: src/Test.FunctionalTests.BinSkim.Driver/BaselineTestsData/Expected/Binskim.linux-x64.dll.sarif [hunk 1]
```diff
@@ -373,7 +373,7 @@
             "id": "NotApplicable_InvalidMetadata",
             "arguments": [
               "Binskim.linux-x64.dll",
-              "EnableAdditionalSdlSecurityChecks",
+              "EnableMicrosoftCompilerSdlSwitch",
```
Comment (marmegh) [inline src/Test.FunctionalTests.BinSkim.Driver/BaselineTestsData/Expected/Binskim.linux-x64.dll.sarif:376]: >"EnableMicrosoftCompilerSdlSwitch", [](http://example.com/codeflow?start=14&length=35)

The only change for all of these sarif results is for the rule name change.

**Comment by marmegh** [comment]
Please note, you will see a list of executables that have moved. This is due to the folder being renamed to match the new rule friendly name.

File: src/Test.FunctionalTests.BinSkim.Driver/BaselineTestsData/Expected/Binskim.linux-x64.dll.sarif (hunk 1 above)
Comment (marmegh) [reply src/Test.FunctionalTests.BinSkim.Driver/BaselineTestsData/Expected/Binskim.linux-x64.dll.sarif:376]: Many of these sarif file changes will become obsolete if #590 is completed first.

File: src/BinSkim.Rules/RuleIds.cs [hunk 2]
```diff
@@ -40,7 +40,7 @@ internal static class RuleIds
 
         public const string EnableSpectreMitigations = "BA2024";
         public const string EnableShadowStack = "BA2025";
-        public const string EnableAdditionalSdlSecurityChecks = "BA2026";
+        public const string EnableMicrosoftCompilerSdlSwitch = "BA2026";
```
Comment (michaelcfanning) [inline src/BinSkim.Rules/RuleIds.cs:43]: >EnableMicrosoftCompilerSdlSwitch [](http://example.com/codeflow?start=28&length=32)

// Renamed from 'EnableAdditionalSdlSecurityChecks' as of 1.9.3.

**Review by michaelcfanning — APPROVED** [review]
:shipit:

**Comment by eddynaka** [comment]
can u fix the conflicts, pls?

**Pull request merged**